apps/test/util/testUtils.js

```diff
@@ -132,6 +133,40 @@ exports.runOnAppTick = function (app, tick, fn) {
 };
 
 /**
+ * Check a given predicate every tick, returning a promise that will resolve
+ * when the predicate first evaluates TRUE.  This provides an easy way to
+ * wait for certain asynchronous test setup to complete before checking
+ * assertions.
+ *
+ * @param {Studio|Applab|GameLab} app
+ * @param {function} predicate
+ * @returns {Promise} to resolve when predicate is true
+ *
+ * @example
+ *   tickAppUntil(Applab, function () {
+ *     return document.getElementById('slow-element');
+ *   }).then(function () {
+ *     assert(document.getElementById('slow-element').textContent === 'pass');
+ *   });
+ */
+exports.tickAppUntil = function (app, predicate) {
+  if (!app) {
+    throw new Error('not supported outside of studio/applab');
```

Comment:
This conditional just checks that I pass something, not that this something is studio/applab. I wonder if the check should be
```
if (!app || !app.onTick)
```